### ISSUE TITLE
Ruby MessageExts should define to_json with an `options` parameter

### ISSUE THREAD

**Issue by jmillikin-stripe** [protocolbuffers/protobuf#3721]
The Ruby standard JSON module assumes `to_json` methods have a single positional parameter for propagating serialization options down the call stack. The protobuf `MessageExts` module ([message_exts.rb](https://github.com/google/protobuf/blob/master/ruby/lib/google/protobuf/message_exts.rb)) defines `to_json` with no parameters, which causes errors when serializing containers:

```ruby
# $ ruby test_json.rb 
# test_json.rb:3:in `to_json': wrong number of arguments (1 for 0) (ArgumentError)
# 	from test_json.rb:7:in `to_json'
# 	from test_json.rb:7:in `<main>'
require 'json'
class Foo 
  def to_json
    'FOO'
  end 
end
puts [Foo.new].to_json

# $ ruby test_json.rb 
# [FOO]
require 'json'
class Foo
  def to_json(opts)
    'FOO'
  end
end
puts [Foo.new].to_json
```

**Comment by blowmage** [comment]
Was this fixed by #4252?

**Comment by jmillikin-stripe** [comment]
Yes, that PR looks like it fixes this ticket.
